### PR TITLE
CI: implement job skipping in Python matrix calculation

### PR DESCRIPTION
**Pull request by Kobzol**
This removes the `step` YAML anchor and the corresponding bash script.

Best reviewed commit-by-commit.

r? @pietroalbini

### CODE REVIEW

**Comment by Kobzol** [comment]
@bors r-

(found an error in one of the commits)

File: src/ci/github-actions/jobs.yml [hunk 1]
```diff
@@ -200,24 +200,23 @@ auto:
...
   - image: x86_64-gnu-integration
-    env:
-      # Only run this job on the nightly channel. Fuchsia requires
-      # nightly features to compile, and this job would fail if
-      # executed on beta and stable.
-      CI_ONLY_WHEN_CHANNEL: nightly
+    # Only run this job on the nightly channel. Fuchsia requires
+    # nightly features to compile, and this job would fail if
+    # executed on beta and stable.
+    ci_only_when_channel: nightly
```
Comment (pietroalbini) [inline src/ci/github-actions/jobs.yml:219]: Bikeshed: if it's not an environment variable anymore I'd rename it to `only-on-channel` or something like that, we don't need the `ci_` prefix (I don't have an opinion on `-` vs `_` though).

Comment (Kobzol) [reply src/ci/github-actions/jobs.yml]: We already use underscores everywhere else in the document, so let's keep that for consistency. I changed it to `only_on_channel`, good point.

**Pull request merged**